### DISCUSSION TITLE
MIDI stop control

### DISCUSSION THREAD

**Discussion by peterthesteam** [GrandOrgue/grandorgue#1075] [Help]
Hi all
I am a retired organ builder of mechanical organs - Street organs and Fairground organs. I have been making an organ up in GO using samples of pipework I have built. Complex, as I am building it on the scale of a cardboard music book. And so the organ is not chromatic. So far I have managed to get it going quite successfully, however I have ground to a halt with controlling the stops from the keyboard.
In the cardboard music there is one key which shuts all the stops off, however there are longer slots which override this to keep those particular stops latched on. Now I assume that I can do something similar by having a general cancel button, and holding the stops 'on' that I require. However I can't work out how to operate the stops or the general cancel button from any keys of the keyboard - in other words the same MIDI channel as the rest of the notes. Any tips and hints would be gratefully received.
Peter

**Comment by larspalo** [comment]
Just right click on the element on screen that you want to control and select Listen for event in the dialog, push the MIDI keyboard key that you want to use for controlling that element. There are multiple ways to toggle the stops on/off too in that dialog that you can explore.

The combination setter allows for programming any kind of combinations, and you can also use the generals for that.

**Reply by peterthesteam** [reply]
Thanks for the reply larspalo. 
My problem has been that I tried the 'listen for event', and the program seemed to accept that. However I still couldn't get the stop to operate from the key. I have set up a General Cancel button, and will now see if that works from the keyboard.
Peter

**Reply by larspalo** [reply]
What type of event did you select for it in the dialog?

**Reply by peterthesteam** [reply]
Hi Lars
I have got it working!! I have discovered that Listen For Event does not work when playing a note from a MIDI file through the internal player. Neither does it work when selecting the note from the keyboard on the screen. 
So I manually selected the Channel and the Data number with a Note On Toggle. I then discovered that the data number is MIDI starting at 0 rather than being the same as the keyboards starting on MIDI 1. That confused me for a while.
Anyway, I now have one 'note' that works a general cancel, and then one note for each stop. This is an odd arrangement for a church organ console, but is needed for me to emulate 'Book' music. I now have a virtual Black Forest Fairground Organ playing.

Thank you for your help.
Peter
